### ISSUE TITLE
 .geo file or other CAD model file To .dmg file

### ISSUE THREAD

**Issue by cenxin995886** [SCOREC/core#196]
Hi,I wanted to use GMSH to generate the initial grid and then use PUMI for parallel grid partitioning, refinement, and more.
But PUMI only use `.smb` and `.dmg`, PUMI could transport `.msh` file to `.smb` file through `from_gmsh`.But in this way,I have to use gmodel to create geometric model and get `.geo`file  `.dmg` file .It's very inconvenient,It is certainly not possible to create complex geometric models.So I want to do that PUMI could convert `.geo` file or other CAD model files directly to .dmg file.In this way, I can directly create geometric models in GMSH and CAD modeling software.
In summary, I wanted to write an interface that allows PUMI to read `.geo` files in GMSH and other CAD model files and convert them to `.dmg` files.
So,would you give me some ideas?where should I have a look ,where could I reference to study? 
Thany you !

**Comment by jacobmerson** [comment]
Do you have access to Simmetrix? `simTranslate` can be used to convert a cad file to a simmetrix `.smd` file, then `mdlConvert` can be used to convert this into a pumi `.dmg` model.

Please see: https://github.com/SCOREC/core/wiki/Geometric-Models

This might be a good starting point.

**Comment by jacobmerson** [comment]
You might be able to write a `.geo` reader in GModel. @ibaned and @cwsmith might have something more to say on this issue.

**Comment by ibaned** [comment]
`.geo`'s syntax is non-trivial, but it could be done in theory. I have code in another project for building general parsers in C++, so that could help.

**Comment by cenxin995886** [comment]
Hi,I don't have to convert from `.geo` model files, other general CAD model files are also available, such as `.iges`, `.step`. So could you provide me with the code connection or file for the project you are working on?thank you. @ibaned

**Comment by cwsmith** [comment]
I'm closing this issue.  Please reopen if needed.
